src/lib/reasoners/sat_solver_sig.mli

```diff
@@ -28,6 +28,9 @@
 (*                                                                        *)
 (**************************************************************************)
 
+(** A reason for which the solver returned unknown. *)
+type unknown_reason = string
```

Comment:
> Nit: It would be cleaner for this to be an ADT, but we can keep a string for now (although I think the string is always `"incomplete"` currently)

I added this comment in sat_solver_sig.ml

Reply:
After merging #829, the ADT will be something as 
```ocaml 
type unknown_reason = 
  | Incomplete 
  | Memout 
  | Timeout of timeout_reason
``` 
We can add the ADT right now but we will have to modify it just after merging the other PR.

We can do it in this PR but we will have to modify the ADT just after merging O

Reply:
We merged #829 so we can create the ADT now.

Reply:
This ADT is strange, it makes the possibility of building a value `Timeout NoTimeout`. I'm changing this a bit.